client/lxd_instances.go

```diff
@@ -1245,27 +1244,39 @@ func (r *ProtocolLXD) rawSFTPConn(apiURL *url.URL) (net.Conn, error) {
 	req.Header["Upgrade"] = []string{"sftp"}
 	req.Header["Connection"] = []string{"Upgrade"}
 
+	// Set the user agent.
+	if r.httpUserAgent != "" {
+		req.Header["User-Agent"] = []string{r.httpUserAgent}
+	}
+
+	if r.requireAuthenticated {
+		req.Header["X-LXD-authenticated"] = []string{"true"}
+	}
+
+	// Set macaroon headers if needed.
+	if r.bakeryClient != nil {
+		r.addMacaroonHeaders(req)
+	}
+
+	// Establish the connection.
 	conn, err := httpTransport.Dial("tcp", apiURL.Host)
 	if err != nil {
 		return nil, err
 	}
 
-	tcpConn, err := tcp.ExtractConn(conn)
-	if err == nil {
-		err = tcp.SetTimeouts(tcpConn)
+	if httpTransport.TLSClientConfig != nil {
+		conn, err = httpTransport.DialTLS("tcp", apiURL.Host)
```

Comment:
@stgraber I'll merge this anyway, as it clearly fixes some important auth issues I'd not considered.
However I'd like to follow this up with another PR that avoids needing to dialling the remote LXD server twice (firstly with httpTransport.Dial on line 1262, and then again (if TLS) on line 1268 with httpTransport.DialTLS).

Can we change this to be:

```go
var conn net.Conn

if httpTransport.TLSClientConfig != nil {
    conn, err = httpTransport.DialTLS("tcp", apiURL.Host)
    if err != nil {
        return nil, err
    }
} else {
    conn, err = httpTransport.Dial("tcp", apiURL.Host)
    if err != nil {
         return nil, err
    }
}
```